client/src/components/body/profilepage/Labs.js

```diff
@@ -1,47 +1,188 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import LabGeneration from "../lab/LabGeneration";
 import useMainStateContext from "src/reducers/MainContext";
 import PropTypes from "prop-types";
+import { Input } from "reactstrap";
 
 const Labs = (props) => {
   const { actions, state } = useMainStateContext();
+
+  const [currentHeader, setCurrentHeader] = useState("");
+
+  const displayNotStartedLabs = () => {
+    setCurrentHeader("Not Started");
+    return (
+      <LabGeneration
+        actions={actions}
+        progressState="NOT_STARTED"
+        labids={props.toDoLabs}
+      />
+    );
+  };
+
+  const [search, setSearch] = useState("");
+  const [displayedLabs, setDisplayedLabs] = useState(displayNotStartedLabs);
+
+  useEffect(() => {
+    setDisplayedLabs(displayNotStartedLabs());
+  }, [props.toDoLabs]);
+
+  const displayInProgressLabs = () => {
+    setCurrentHeader("In Progress");
+    return (
+      <LabGeneration
+        actions={actions}
+        progressState="IN_PROGRESS"
+        labRecords={props.inProgressLabs}
+      />
+    );
+  };
+
+  const displayCompletedLabs = () => {
+    setCurrentHeader("Completed");
+    return (
+      <LabGeneration
+        actions={actions}
+        progressState="COMPLETED"
+        labRecords={props.completedLabs}
+      />
+    );
+  };
+
+  const searchLabs = (e) => {
+    e.preventDefault();
+    let searchResults = {
+      notStarted: [],
+      inProgress: [],
+      completed: [],
+    };
+
+    props.toDoLabs?.forEach((lab) => {
+      if (lab.labName.toLowerCase().includes(search.toLowerCase())) {
+        searchResults.notStarted.push(lab);
+      }
+    });
+
+    props.inProgressLabs?.forEach((lab) => {
+      if (lab.labName.toLowerCase().includes(search.toLowerCase())) {
+        searchResults.inProgress.push(lab);
+      }
+    });
+
+    props.completedLabs?.forEach((lab) => {
+      if (lab.labName.toLowerCase().includes(search.toLowerCase())) {
+        searchResults.completed.push(lab);
+      }
+    });
+
+    const displaySearchResults = (
+      <div>
+        <LabGeneration
+          actions={actions}
+          progressState="NOT_STARTED"
+          labids={searchResults.notStarted}
+          search={true}
+        />
+        <LabGeneration
+          actions={actions}
+          progressState="IN_PROGRESS"
+          labRecords={searchResults.inProgress}
+          search={true}
+        />
+        <LabGeneration
+          actions={actions}
+          progressState="COMPLETED"
+          labRecords={searchResults.completed}
+          search={true}
+        />
```

Comment:
![Screenshot 2024-10-25 at 1 18 41 PM](https://github.com/user-attachments/assets/f6cac522-6df8-4de3-aea6-288f44db9264)
The spacing looks weird here - why is there a big gap?

Reply:
![Screenshot 2024-10-25 at 1 20 05 PM](https://github.com/user-attachments/assets/baf91a2d-091f-4c49-8b25-9755db6d6a0e)
Spacing here looks weird too

Reply:
![Screenshot 2024-10-25 at 1 21 25 PM](https://github.com/user-attachments/assets/17d2c174-c6e2-4382-8af5-7007a2d5905c)
Here too. Make sure you're using tailwind for mobile design first, then define the styling for md and lg screens

Reply:
Touched base and will move forward with not started over assigned labs